UKHomeOffice/sgmr-service
# SPCR-264 Onboarding flow

### AC / Description of changes

- add skipper section to onboarding
- add people section to onboarding


### To Test

- ensure your account has no people, no voyages, and no pleasure craft saved
- go to dashboard
- > you should see the onboarding banner
- click start
- > you should be taken to the pleasure craft form
- add a pleasure craft and save
- > you should be taken to the 'are you the skipper question'
- click no
- > you should be taken to the 'do you want to add people page'
- use the browser back to return to the skipper question
- click yes
- > you should be taken to the people form 
- complete the form
- click save
- > you should be taken to the 'do you want to add people' page
- click yes
- > you should be taken to the people form
- add a person
- click save
- > you should be returned to the 'do you want to add people' page
- click no
- > you should be returned to the dashboard
- > there should be no onboarding component


### Notes

While we take the 'skipper' information we just create them as a regular person, we do not have the concept of a saved skipper yet

To make testing easier, rather than clearing your account of all voyages, people, and pleasurecraft you can set the onboarding state to always be true

- on Dashboard.jsx
- within the useEffect starting on line 73
- add setShowOnboarding(true)

```
useEffect(() => {
    if (userData && Object.keys(voyageStatusCounts).length > 0) {
      const totalVoyages = Object.values(voyageStatusCounts).reduce((a, b) => a + b);
      setShowOnboarding(userData.people.length === 0 && userData.vessels.length === 0 && totalVoyages === 0);
    }
    setShowOnboarding(true);
  }, [userData, voyageStatusCounts, setShowOnboarding]);
  ```

---
* remember to merge to feature branches not master *
